token-lending/program/tests/liquidate.rs

```diff
@@ -156,4 +204,28 @@ async fn test_success() {
         sol_liquidated,
         sol_loan_state.borrowed_liquidity_wads.round_u64()
     );
+
+    let collateral_liquidated =
+        SOL_LOAN_USDC_COLLATERAL - sol_loan_state.deposited_collateral_tokens;
+    assert!(collateral_liquidated > 0)
+}
+
+#[tokio::test]
+async fn test_liquidate_healthy_obligation_failure() {
+    let TestReturn { usdc_result, .. } = setup(100, 100, false).await;
```

Comment:
My first real contribution. I rewired the liquidate function slightly and used it to actually test that one of your validations works! What do you think?

Reply:
I definitely think as I grow this test file I will need to re-parameterize setup to take a struct instead of comma separated parameters because it'll get too unwieldy.

Reply:
Also will want to tell it whether I want a USDC loan or a SOL loan, not both...

Reply:
> My first real contribution

haha, woot!

> I will need to re-parameterize setup to take a struct instead of comma separated parameters because it'll get too unwieldy. Also will want to tell it whether I want a USDC loan or a SOL loan, not both...

Those changes sound good to me